lib/zendesk_apps_support/package.rb

```diff
@@ -212,6 +215,13 @@ def has_requirements?
       has_file?(REQUIREMENTS_FILENAME)
     end
 
+    def self.has_custom_object_requirements?(requirements_hash)
+      return false if requirements_hash.nil?
+      types = requirements_hash.dig(CUSTOM_OBJECTS_REQUIREMENT_KEY, CUSTOM_OBJECTS_REQUIREMENT_TYPE_KEY) || []
+      relationships = requirements_hash.dig(CUSTOM_OBJECTS_REQUIREMENT_KEY, CUSTOM_OBJECTS_RELATIONSHIP_TYPE_KEY) || []
+      types.any? || relationships.any?
```

Comment:
This already looks a whole lot better than `a && b || c`. Well done. 

Wanted to point out that line 222 can be refactored to `(types | relationships).any?`
One can argue that this reduces readability 

Reference. http://www.chrisrolle.com/en/blog/array-coherences